src/libsync/propagatedownload.cpp

```diff
@@ -451,19 +436,69 @@ void PropagateDownloadFile::startDownload()
     {
         SyncJournalDb::DownloadInfo pi;
         pi._etag = _item->_etag;
-        pi._tmpfile = tmpFileName;
+        pi._tmpfile = _tmpFileName;
         pi._valid = true;
         propagator()->_journal->setDownloadInfo(_item->_file, pi);
         propagator()->_journal->commit("download file start");
     }
 
+    if (isZsyncPropagationEnabled(propagator(), _item)) {
+        if (_item->_previousSize) {
+            QNetworkRequest req;
+            req.setPriority(QNetworkRequest::LowPriority);
+
+            QUrl zsyncUrl = Utility::concatUrlPath(propagator()->account()->zsyncUrl(),
+                                                   propagator()->_remoteFolder + _item->_file);
+            auto job = propagator()->account()->sendRequest("GET", zsyncUrl, req);
+            connect(job, &SimpleNetworkJob::finishedSignal, this, &PropagateDownloadFile::slotZsyncGetMetaFinished);
+            return;
+        }
+        qCInfo(lcZsyncGet) << "No local copy of:" << propagator()->getFilePath(_item->_file);
+    }
+
+    startDownloadNormal();
+}
+
+
+void PropagateDownloadFile::slotZsyncGetMetaFinished(QNetworkReply *reply)
+{
+    int httpStatusCode = reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
+    if (httpStatusCode != 200) {
+        qCInfo(lcPropagateDownload)
+            << "No remote zsync file for:"
+            << propagator()->getFilePath(_item->_file);
+        startDownloadNormal();
+        return;
+    }
+
+    QByteArray zsyncData = reply->readAll();
+
+    qCInfo(lcZsyncGet) << "Got remote zsync file for:" << _item->_file << "size:" << zsyncData.size();
+
+    QMap<QByteArray, QByteArray> headers;
+    _job = new GETFileZsyncJob(propagator()->account(), _item, propagator(),
+                               propagator()->_remoteFolder + _item->_file,
+                               propagator()->getFilePath(_item->_file),
+                               propagator()->getFilePath(_tmpFileName),
+                               &_tmpFile, headers, _expectedEtagForResume, zsyncData, this);
+    connect(_job.data(), &GETJob::finishedSignal, this, &PropagateDownloadFile::slotGetFinished);
+    connect(qobject_cast<GETFileZsyncJob*>(_job.data()), &GETFileZsyncJob::downloadProgress,
```

Comment:
Done.

Reply:
That means `GETFileZsyncJob` redeclares a different `downloadProgress` -- why?

Reply:
It is a bit confusing but it was the only way I could get `GETFileZsyncJob` to send correct progress. See `GETFileZsyncJob::slotDownloadProgress`. If I'm missing something in terms of a better way to do this please let me know.

Reply:
Ah, I see. It's because GETFileZsyncJob is actually multiple chained jobs and you want to report overall progress and not per-job progress. Suggestion: Rename the signal to "overallProgress"?

I think that `GETFileZsyncJob` isn't a `AbstractNetworkJob` at all and that's where the confusion comes from. It *has* multiple of them. Refactoring `GETJob` to not be `ANJ` derived would probably be significant work though.

Reply:
Rename to `::slotOverallDownloadProgress` done.

Reply:
Done.